### PR TITLE
MOSIP-26334 Fixed update uin invalid lang code error.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 1]
```diff
@@ -844,6 +846,28 @@ public void validateUpdateRequest(RequestWrapper<ResidentUpdateRequestDto> reque
 		}
 	}
 
+	private void validateLanguageCodeInIdentityJson(JSONObject identity) {
+		// Get a set of entries
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Add a null check for identity

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 2]
```diff
@@ -844,6 +846,28 @@ public void validateUpdateRequest(RequestWrapper<ResidentUpdateRequestDto> reque
...
+		for (Map.Entry entry : (Iterable<Map.Entry>) identity.entrySet()) {
+			// Retrieve the key and value of each entry
+			String key = (String) entry.getKey();
+			Object value = entry.getValue();
+			if (value instanceof ArrayList){
+				ArrayList valueArray = (ArrayList) value;
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Make the type as `ArrayList<?>

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 3]
```diff
@@ -844,6 +846,28 @@ public void validateUpdateRequest(RequestWrapper<ResidentUpdateRequestDto> reque
...
+				for(Object valueInList: valueArray){
+					if(valueInList instanceof LinkedHashMap){
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Check for instanceof Map but not LinkedHashMap

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 4]
```diff
@@ -844,6 +846,28 @@ public void validateUpdateRequest(RequestWrapper<ResidentUpdateRequestDto> reque
...
+						LinkedHashMap<String, String> valueInListMap = (LinkedHashMap<String, String>) valueInList;
+						if(valueInListMap.containsKey(ResidentConstants.LANGUAGE)){
+							validateLanguageCode(valueInListMap.get(ResidentConstants.LANGUAGE));
+						}
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java:869]: * we should also check "language" attribute not null and not empty.
* And we should also check "value" attribute not null and not empty.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Done

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 4 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java:869]: For "language" attribute null check is already taken care of in validateLanguageCode() method.
for the value I have checked.

Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java:869]: For "language" attribute null check is already taken care of in validateLanguageCode() method.
for the value, I have checked.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 5]
```diff
@@ -844,6 +846,34 @@ public void validateUpdateRequest(RequestWrapper<ResidentUpdateRequestDto> reque
...
+				if (value instanceof ArrayList<?>) {
+					ArrayList<?> valueArray = (ArrayList<?>) value;
+					for (Object valueInList : valueArray) {
+						if (valueInList instanceof Map) {
+							Map <String, String> valueInListMap = (Map <String, String>) valueInList;
+							if (valueInListMap.containsKey(ResidentConstants.LANGUAGE)) {
+								String languageCode = valueInListMap.get(ResidentConstants.LANGUAGE);
+								validateLanguageCode(languageCode);
+								String valueOfLanguageCode = valueInListMap.get(ResidentConstants.VALUE);
+								if(valueOfLanguageCode==null || valueOfLanguageCode.isEmpty()){
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: for null or empty value of "language" and "value" throw missing input parameter exception

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 5 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Done.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**